arcgis-ios-sdk-samples.xcodeproj/project.pbxproj

```diff
@@ -5049,6 +5072,24 @@
 /* End PBXResourcesBuildPhase section */
 
 /* Begin PBXShellScriptBuildPhase section */
+		007E8EF32578290F003077CC /* Create .secrets file if it does not exist */ = {
+			isa = PBXShellScriptBuildPhase;
+			buildActionMask = 2147483647;
+			files = (
+			);
+			inputFileListPaths = (
+			);
+			inputPaths = (
+			);
+			name = "Create .secrets file if it does not exist";
+			outputFileListPaths = (
+			);
+			outputPaths = (
+			);
+			runOnlyForDeploymentPostprocessing = 0;
+			shellPath = /bin/sh;
+			shellScript = "if [ ! -e \"$SRCROOT/.secrets\" ]\nthen\n    touch \"$SRCROOT/.secrets\"\nfi\n";
```

Comment:
Hi @philium ,

![image](https://user-images.githubusercontent.com/9660181/100944794-ce031c00-34b4-11eb-91f3-aed7bc92f6ca.png)

Currently when we build from the branch, it shows `error: Build input file cannot be found: `. I think it is not ideal to have a build error when a user cloned the repo and build for the first time, without setting up the `.secrets` file. 

So I made these changes: 

- if `.secrets` doesn't exist, create a placeholder with this build phase script. 
- I also hardcoded the path to the `.secrets` file in the script to avoid discrepancy between build phase and build rule

In this way, even if the first time users don't know/care how to set up the keys, they can still build and run the app, only to see the authentication challenge when they open a sample with the new basemap constructors.

What's your thought?

p.s. Sorry for keeping changing the PR, as we gradually learn from the discussions with other platforms. 😳

Reply:
Nice!

Reply:
After digging it up, I found the more appropriate place for this script is in Schemes (.xcscheme) -> Pre Actions. In this way, the script will run before the build process starts, and we do not need to hardcode the path in the build rule.